src/cdk-experimental/menu/menu-base.ts

```diff
@@ -157,31 +183,51 @@ export abstract class CdkMenuBase
         startWith(this.items),
         mergeMap((list: QueryList<CdkMenuItem>) =>
           list
-            .filter(item => item.hasMenu())
+            .filter(item => item.hasMenu)
             .map(item => item.getMenuTrigger()!.opened.pipe(mapTo(item), takeUntil(exitCondition))),
         ),
         mergeAll(),
         switchMap((item: CdkMenuItem) => {
-          this.openItem = item;
+          this.triggerItem = item;
           return item.getMenuTrigger()!.closed;
         }),
         takeUntil(this.destroyed),
       )
-      .subscribe(() => (this.openItem = undefined));
+      .subscribe(() => (this.triggerItem = undefined));
   }
 
-  /** Subscribe to the MenuStack close and empty observables. */
+  /** Subscribe to the MenuStack close events. */
   private _subscribeToMenuStackClosed() {
     this.menuStack.closed
       .pipe(takeUntil(this.destroyed))
-      .subscribe(({item, focusParentTrigger}) => this.closeOpenMenu(item, focusParentTrigger));
+      .subscribe(({item, focusParentTrigger}) => this.closeOpenMenu(item, {focusParentTrigger}));
   }
 
-  private _subscribeToHasFocus() {
-    if (this._isInline) {
+  /** Subscribe to the MenuStack hasFocus events. */
+  private _subscribeToMenuStackHasFocus() {
+    if (this.isInline) {
       this.menuStack.hasFocus.pipe(takeUntil(this.destroyed)).subscribe(hasFocus => {
-        this._hasFocus = hasFocus;
+        this._calculateTabIndex(hasFocus);
+      });
+    }
+  }
+
+  /** Calculate the tabindex for the menu host element. */
+  private _calculateTabIndex(menuStackHasFocus: boolean) {
+    const tabindexIfInline = menuStackHasFocus ? -1 : 0;
+    this._tabIndex = this.isInline ? tabindexIfInline : null;
```

Comment:
Should we just have a getter function for this? This seems brittle since you'd have to remember to call it when `isInline` changes.

Reply:
Yeah I guess, I wonder if we should just make `isInline` readonly. I already made if readonly on the concrete classes, I just left it writeable here in case someone wanted to be able to write it. I can't really think of a use case where someone would need to do that though. Anyways for now I'll just make it a getter, this logic is super lightweight anyways